wdl/src/main/scala/wdl/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -176,6 +176,23 @@ trait WdlStandardLibraryFunctions extends WdlFunctions[WomValue] {
     extractArguments flatMap arrayLength
   }
 
+  def flatten(params: Seq[Try[WomValue]]): Try[WomValue] = {
+    def getFlatValues(v: WomValue): List[WomValue] = v match {
+      case WomArray(_, values) => values.toList
+      case other => throw new IllegalArgumentException(s"Invalid argument to flatten(): ${other}, flatten() takes an array of arrays")
+    }
+
+    val arg: Try[WomValue] = extractSingleArgument("flatten", params)
+    arg flatMap {
+      case WomArray(WomArrayType(WomArrayType(elemType)), arrayValues) =>
+        Try {
+          WomArray(WomArrayType(elemType), arrayValues.flatMap(getFlatValues))
```

Comment:
Rather than throwing an exception in `getFlatValues` and wrapping this in a `Try {}`, can we instead have `getFlatValues` return a `Try` and let this use `TryUtils.sequence` to get the result from the `List[Try[_]]`?

Reply:
"TryUtils.sequence"

*sigh*

Reply:
For the life of me, I could not figure out how to do this. Seems like `TryUtils.sequence` is the secret sauce.

Reply:
FWIW elsewhere we've moved away from `Try` to `Validated` which gets this kind of composition for free, but the WDL expression code is still in `Try`land, so `TryUtils` it is!